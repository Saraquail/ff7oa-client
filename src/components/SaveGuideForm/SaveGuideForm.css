
.modal {
  position: absolute;
  top: 50%;
  left: 50%;
  display: flex;
  padding: 2em;
  min-width: 10em;
  max-width: 70%;
  min-height: 10em;
  max-height: 30em;
  background-image: linear-gradient(to bottom right, rgba(14,51,95,.9), rgba(4, 3, 44, 0.9));
  border-radius: 1em;
  transform: translate(-50%, -50%);
  border: solid 1px #e7dfe7;
  box-shadow: 1px 1px #e7dfe7,
              -1px -1px #e7dfe7,
              1px -1px #e7dfe7,
              -1px 1px #e7dfe7,
              -2px 0 #7b757b,
              0 2px #424542;
}

.overlay {
  position: fixed;
  top: 50%;
  right: 0;
  bottom: 0;
  left: 50%;
  color: #4dbdd6;
  font-weight: 700;
}

.overlay {
  display: flex;
  flex-direction: column;
  margin: 0;
}
/* 
.add-guide-button {
  background-color: #04009d;
  margin-bottom: .5em;
  margin-top: 1em;
  color: white;
  font-size: 14px;
} */

#add-guide-form input {
  margin: auto;
  margin-bottom: .5em;
  display: block;
}

