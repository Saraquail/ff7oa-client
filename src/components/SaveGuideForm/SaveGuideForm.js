import React, { Component } from 'react'
import TokenService from '../../services/token-service'
import MonsterApiService from '../../services/monster-api-service'
import './SaveGuideForm.css'


class SaveGuideForm extends Component {
  state = {
    nickname: '',
    note: ''
  }

  clearForm = () => {
    document.getElementById("add-guide-form").reset();
  }

  handleSubmit = ev => {
    ev.preventDefault()
    let id = this.props.selectedid
    let name = this.state.nickname
    let note = this.state.note
    const user_name = TokenService.getUserName()

    const guide = {
      monster_id: id,
      name: name, 
      note: note, 
      user_name: user_name
    }

    MonsterApiService.postGuide(user_name, guide)
      .then(this.clearForm)
      .then(this.props.handleCloseModal)
  }

  handleInputChange = (ev) => {
    ev.preventDefault()

    const target = ev.target;
    const value = target.value 
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  render () {
    return (
      <div className="overlay-container">
        <div className="overlay modal" >
          <form name="add-guide-form" id="add-guide-form" onSubmit={this.handleSubmit}>
            <label htmlFor="nickname">Nickname</label>
              <input type="text" name="nickname" id="nickname" required onChange={this.handleInputChange}/>
            <label htmlFor="note">Note</label>
              <input type="text" name="note" id="note" required onChange={this.handleInputChange}/>
            <button className="add-guide-button" onClick={this.props.handleCloseModal}>Not Interested</button>
            <button className="add-guide-button" type="submit">OK, add it</button>
          </form>
        </div>
      </div>
    )
  }
}

export default SaveGuideForm