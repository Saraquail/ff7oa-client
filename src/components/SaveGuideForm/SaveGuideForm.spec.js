import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import SaveGuideForm from './SaveGuideForm.js';

describe('<SaveGuideForm/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <SaveGuideForm />
      </BrowserRouter>, div,
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SaveGuideForm />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
