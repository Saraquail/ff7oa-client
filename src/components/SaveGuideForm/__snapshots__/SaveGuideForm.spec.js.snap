// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveGuideForm/> renders the UI as expected 1`] = `
<div
  className="overlay-container"
>
  <div
    className="overlay modal"
  >
    <form
      id="add-guide-form"
      name="add-guide-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="nickname"
      >
        Nickname
      </label>
      <input
        autoFocus={true}
        id="nickname"
        name="nickname"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        htmlFor="note"
      >
        Note
      </label>
      <input
        id="note"
        name="note"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <button
        className="add-guide-button"
        type="submit"
      >
        OK, add it
      </button>
      <button
        className="add-guide-button"
        type="button"
      >
        Not Interested
      </button>
      
    </form>
  </div>
</div>
`;
